src/iceberg_destination.rs

```diff
@@ -0,0 +1,201 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use arrow::array::RecordBatch;
+use arrow_schema::{Field, Schema, SchemaRef};
+use futures::{pin_mut, StreamExt, TryStream};
+use iceberg::io::FileIO;
+use iceberg::spec::{
+    DataContentType, DataFileFormat, FormatVersion, Manifest, ManifestContentType, ManifestEntry,
+    ManifestFile, ManifestListWriter, ManifestMetadata, ManifestStatus, ManifestWriter, Operation,
+    PartitionSpec, Snapshot, Struct, Summary, TableMetadata, TableMetadataBuilder,
+};
+use iceberg::writer::file_writer::location_generator::{
+    DefaultFileNameGenerator, DefaultLocationGenerator,
+};
+use iceberg::writer::file_writer::{FileWriter, FileWriterBuilder, ParquetWriterBuilder};
+use iceberg::TableCreation;
+use log::info;
+use parquet::arrow::PARQUET_FIELD_ID_META_KEY;
+use parquet::file::properties::WriterProperties;
+use url::Url;
+use uuid::Uuid;
+
+use crate::error::DataLoadingError;
+
+pub async fn record_batches_to_iceberg(
+    record_batch_stream: impl TryStream<Item = Result<RecordBatch, DataLoadingError>>,
+    arrow_schema: SchemaRef,
+    target_url: Url,
+) -> Result<(), DataLoadingError> {
+    pin_mut!(record_batch_stream);
+
+    let mut file_io_props: Vec<(String, String)> = vec![];
+    if let Ok(aws_endpoint) = std::env::var("AWS_ENDPOINT") {
+        file_io_props.push(("s3.endpoint".to_string(), aws_endpoint));
+    }
+
+    let file_io = FileIO::from_path(target_url.clone())?
+        .with_props(file_io_props)
+        .build()?;
+
+    // Assign sequential field IDs starting from 1 to the fields
```

Comment:
A comment of a block like this usually indicates a good point where the code can be moved to a function: 

`fn assign_sequential_field_ids(...)`